block_iagora.php

```diff
@@ -49,17 +49,34 @@ public function get_content() {
         $this->content = new stdClass();
         $this->content->footer = '';
 
-        $iframeurl = isset($this->config->iframeurl) ? $this->config->iframeurl : '';
+        $copilotendpointurl = isset($this->config->copilotendpointurl) ? $this->config->copilotendpointurl : '';
 
-        if (empty($iframeurl)) {
-            $this->content->text = get_string('noiframeurl', 'block_iagora');
+        if (empty($copilotendpointurl)) {
+            $this->content->text = get_string('nocopilotendpointurl', 'block_iagora');
         } else {
-            $this->content->text = '<iframe src="' . $iframeurl . '" width="100%" height="400px" frameborder="0"></iframe>';
+            $this->content->text = $this->generate_chat_content($copilotendpointurl);
         }
-
+        
         return $this->content;
     }
-
+    
+    /**
+     * Generate the HTML content for the chat block.
+     *
+     * @param string $copilotendpointurl The URL to be used in the iframe.
+     * @return string The generated HTML content for the chat.
+     */
+    private function generate_chat_content($copilotendpointurl) {
+        global $PAGE, $OUTPUT;
+        // Generate a unique identifier for the chat container
+        $chatId = uniqid('iagora_chat_');
+        $context = [
+            'chatId' => $chatId,
+            'tokenEndpointURL' => $copilotendpointurl 
+        ];
+    
+        return $OUTPUT->render_from_template('block_iagora/chat', $context);
+    }
```

Comment:
```suggestion
    }

```
Could we add an empty line between the two functions to visually separate them?

Reply:
Done

Reply:
The empty line is missing:

```php
    private function generate_chat_content($copilotendpointurl) {
      // ....
    }
    /**
     * Define in which pages this block can be added.
```
->
```php
    private function generate_chat_content($copilotendpointurl) {
      // ....
    }

    /**
     * Define in which pages this block can be added.
```